### DISCUSSION TITLE
v4 performance visibly worse than v3

### DISCUSSION THREAD

**Discussion by darkowic** [yarnpkg/berry#6149] [General]
Hi! I just finished migration to yarn v4 and unfortunately performance is worse than in v3:

> 
> Performance change for install packages step (yarn --immutable)
> yarn 3:
> Done with warnings in 1m 32s
> Done with warnings in 1m 28s
> Done with warnings in 1m 24s
> yarn 4:
> Done with warnings in 1m 39s
> Done with warnings in 1m 34s
> Install with freeze not affected much and even slightly slower then before.
> 
> Localhost performance and initial install time from cache:
> yarn 3
> YN0000: Done with warnings in 5m 55s
> YN0000: Done with warnings in 12m 21s
> YN0000: Done with warnings in 5m 53s
> yarn 4 with compression mixed (lvl 9)
> YN0000: · Done with warnings in 6m 43s
> YN0000: · Done with warnings in 8m 9s
> YN0000: · Done with warnings in 9m 1s
> yarn 4 with compression 0
> YN0000: · Done with warnings in 7m 40s
> YN0000: · Done with warnings in 7m 45s
> YN0000: · Done with warnings in 9m 10s
> Actually, yarn 4 is slower in our case than v3.

Definitely its not 3 times faster as release blog post claims it to be. Do you have ideas what happened?
We have plenty of warnings like
> ➤ YN0060: │ eslint is listed by your project with version 8.56.0, which doesn't satisfy what react-scripts (p6eec3) and other dependencies request (^7.5.0).

and our monorepo is quite complex. Link step takes most of the time:
> ➤ YN0000: ┌ Link step
➤ YN0007: │ canvas@npm:2.11.2 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:3.36.0 must be built because it never has been before or the last one failed
➤ YN0007: │ node-sass@npm:9.0.0 must be built because it never has been before or the last one failed
➤ YN0007: │ puppeteer@npm:13.7.0 must be built because it never has been before or the last one failed
➤ YN0007: │ fsevents@patch:fsevents@npm%3A1.2.13#optional!builtin<compat/fsevents>::version=1.2.13&hash=d11327 must be built because it never has been before or the last one failed
➤ YN0007: │ node-sass@npm:8.0.0 must be built because it never has been before or the last one failed
➤ YN0007: │ ejs@npm:2.7.4 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js-pure@npm:3.35.0 must be built because it never has been before or the last one failed
➤ YN0007: │ re2@npm:1.20.3 must be built because it never has been before or the last one failed
➤ YN0007: │ puppeteer@npm:18.1.0 must be built because it never has been before or the last one failed
➤ YN0007: │ @axa-fr/oidc-client@npm:7.19.0 must be built because it never has been before or the last one failed
➤ YN0007: │ puppeteer@npm:15.5.0 must be built because it never has been before or the last one failed
➤ YN0007: │ chromedriver@npm:121.0.2 must be built because it never has been before or the last one failed
➤ YN0007: │ dtrace-provider@npm:0.8.8 must be built because it never has been before or the last one failed
➤ YN0007: │ @nestjs/core@npm:10.3.0 [020c9] must be built because it never has been before or the last one failed
➤ YN0007: │ es5-ext@npm:0.10.62 must be built because it never has been before or the last one failed
➤ YN0007: │ @ffmpeg-installer/darwin-x64@npm:4.1.0 must be built because it never has been before or the last one failed
➤ YN0007: │ esbuild@npm:0.18.20 must be built because it never has been before or the last one failed
➤ YN0007: │ @swc/core@npm:1.4.2 [100ca] must be built because it never has been before or the last one failed
➤ YN0007: │ @openapitools/openapi-generator-cli@npm:2.9.0 must be built because it never has been before or the last one failed
...
➤ YN0000: └ Completed in 7m 29s

**Comment by kopax-polyconseil** [comment]
I am surprised because i just upgrade from yarn 1 to 4 and I cant install.it is just stuck. Only you ask about perf.

Funny. On my phone as yarn4 takes all the ram.
